p2p/src/simulated/network.rs

```diff
@@ -482,6 +494,151 @@ impl crate::Receiver for Receiver {
     type Error = Error;
 
     async fn recv(&mut self) -> Result<Message, Error> {
-        self.receiver.next().await.ok_or(Error::PeerClosed)
+        self.receiver.next().await.ok_or(Error::NetworkClosed)
+    }
+}
+
+// A peer in the simulated network.
+// The peer can receive messages over any registered channel.
+struct Peer {
+    // Map from channel to a sender that can deliver messages to that channel
+    mailboxes: Arc<Mutex<HashMap<Channel, mpsc::UnboundedSender<Message>>>>,
+
+    // Socket address that the peer is listening on
+    socket: SocketAddr,
+}
+
+impl Peer {
```

Comment:
the code in this function gets quite indented, but I couldn't figure out a good way to split it out without putting it into, say, a new function which is invoked only here. So I thought it was marginally more readable to just leave it all in this function (very indented)

Reply:
I think this is fine 👍